### ISSUE TITLE
[Streams] [p]twitch returns wrong error

### ISSUE THREAD

**Issue by fixator10** [Cog-Creators/Red-DiscordBot#2488]
# Command bugs

#### Command name

`twitch`

#### What cog is this command from?

`streams`

#### What were you expecting to happen?

Get "That channel doesn't seem to exist." message

#### What actually happened?

Got "The Twitch token is either invalid or has not been set. See [p]streamset twitchtoken." message
![gif](https://Moeka.is-a-good-waifu.com/6G4th2V.gif)

#### How can we reproduce this issue?

1. Setup api key for twitch
2. Try to invoke cmd `[p]twitch <any-non-ascii-character>` (In example: `[p]twitch пример`)

**Comment by palmtree5** [comment]
For reference, this is the json output when a bad client id is passed:
```json
{"error": "Bad Request", "status": 400, "message": "Invalid client id specified"}
```
and this is the json output when a bad login parameter is passed (using the example of `пример`):
```json
{"error": "Bad Request", "status": 400, "message": "The login parameter is invalid: 'пример'"}
```
Notice that the only difference is in the message parameter. Now for our code:
```py
async with aiohttp.ClientSession() as session:
    async with session.get(url, headers=header) as r:
        data = await r.json()

    if r.status == 200:
        if not data["users"]:
            raise StreamNotFound()
        return data["users"][0]["_id"]
    elif r.status == 400:
        raise InvalidTwitchCredentials()
    else:
        raise APIError()
```
The problem here is that we are not differentiating between causes of a 400 and instead lumping them all together as "Invalid credentials" when that's not necessarily the case.

tl;dr: easy fix, just need to differentiate between causes of a 400
